### PR TITLE
EN-44755: Add type designation system to file format and library and ability to configure a custom application specific crawler-user-agents.json file

### PR DESCRIPTION
**Pull request by housepage**
* Update crawler-user-agents.json file to the latest
* Implement ability of patterns to have a types field that is queryable via bot?(type)
* Implement ability to set VOIGHT_KAMPFF_ROOT and read a custom crawler-user-agents.json file from $VOIGHT_KAMPFF_ROOT/config/crawler-user-agents.json instead of default

### CODE REVIEW

File: lib/tasks/voight_kampff.rake [hunk 1]
```diff
@@ -8,7 +8,7 @@ namespace :voight_kampff do
     uri = URI(args[:url])
     contents = Net::HTTP.get(uri)
 
-    if contents.present?
+    if !contents.nil? && !contents.empty?
```
Comment (QMalcolm) [inline lib/tasks/voight_kampff.rake:11]: Aren't `present?` and this line functionally equivalent? Why do we prefer having it in this expanded form? I just read through https://stackify.com/understanding-absence-in-ruby-present-blank-nil-empty/ to try and understand this better.

Comment (housepage) [reply lib/tasks/voight_kampff.rake:11]: .present? is monkey-patched onto Object by Rails and isn't pure Ruby.

I don't think this library should take a direct Rails/ActiveSupport dependency.

Comment (housepage) [reply lib/tasks/voight_kampff.rake:11]: https://api.rubyonrails.org/classes/Object.html#method-i-present-3F

Comment (QMalcolm) [reply lib/tasks/voight_kampff.rake:11]: Gotcha gotcha

**Review by QMalcolm — APPROVED** [review]
🐑 it

**Review by QMalcolm — APPROVED** [review]
Readme updates make sense. I appreciate the guidance details.

**Pull request merged**